src/getexchange.c

```diff
@@ -615,277 +770,34 @@ void checkexchange(char *comment) {
-
-	if (((comment[0] == 'A')
-		|| (comment[0] == 'B')
-		|| (comment[0] == 'M')
-		|| (comment[0] == 'Q')
-		|| (comment[0] == 'S')
-		|| (comment[0] == 'U'))
-		&& ((comment[1] == ' ') || (cmpattern[2] == 'f'))) {
-
-	    precedent[0] = comment[0];
-	}
-
-
-	/* look for a single letter */
-	for (ii = 0; ii < LEN(precpats); ii++) {
-
-	    hr = getlastpattern(precpats[ii]);
-
-	    if (hr > 0) {
-		pr = comment[hr];
-		if ((pr == 'Q') || (pr == 'A') || (pr == 'B')
-			|| (pr == 'U') || (pr == 'M') || (pr == 'S')) {
-		    precedent[0] = pr;
-		    precedent[1] = '\0';
-		}
-	    }
-	}
-
-	// get call update
-
-	for (ii = 0; ii < LEN(callpats); ii++) {
-
-	    hr = getlastpattern(callpats[ii]);
-
-	    if (hr > 0) {
-		if (((comment[hr] == 'A') && (comment[hr + 1] > 59))
-			|| (comment[hr] == 'K') || (comment[hr] == 'N')
-			|| (comment[hr] == 'W')
-			|| (comment[hr] == 'V') || (comment[hr] == 'C')) {
-
-		    switch (ii) {
-
-			case 0 ... 1:
-			    strncpy(callupdate, comment + hr, 4);
-			    callupdate[4] = '\0';
-			    break;
-			case 2 ... 3:
-			    strncpy(callupdate, comment + hr, 5);
-			    callupdate[5] = '\0';
-			    break;
-			case 4:
-			    strncpy(callupdate, comment + hr, 6);
-			    callupdate[6] = '\0';
-
-		    }
-		    if (strlen(callupdate) > 3) {
-
-			if (call_update == 1)
-			    strcpy(hiscall, callupdate);
-
-			mvprintw(12, 29, "       ");
-			mvprintw(12, 29, "%s", hiscall);
-		    }
-
-		}
-	    }
-
-	}
-
-	// get check
-
-	for (ii = 0; ii < LEN(checkpats); ii++) {
-
-	    hr = getlastpattern(checkpats[ii]);
-	    if (hr > 0) {
-		check[0] = comment[hr];
-		check[1] = comment[hr + 1];
-		check[2] = '\0';
-	    }
-	}
-
-	// get section
-	*section = '\0';
-
-	for (ii = 0; ii < LEN(secpats); ii++) {
-
-	    hr = getlastpattern(secpats[ii]);
-
-	    if (hr > 0) {
-
-		g_strlcpy(checksection, comment + hr, 4);
-		g_strchomp(checksection);
-
-		for (jj = 0; jj < get_mult_count(); jj++) {
-
-		    char *multi = g_strdup(get_mult(jj));
-		    g_strchomp(multi);
-
-		    if ((strlen(multi) >= 1) &&
-			    (strcmp(checksection, multi) == 0)) {
-
-			strcpy(section, multi);
-			break;
-		    }
-		    g_free(multi);
-		}
-	    }
-	}
-
-	{
-	    char buf[40];
-	    sprintf(buf, " %4s %1s %2s %2s ", serial, precedent,
-		    check, section);
-	    OnLowerSearchPanel(8, buf);
-	}
-
-	/* \todo use sprintf */
-	ssexchange[0] = '\0';
+	checkexchange_arrlss(comment, interactive);
+	return;
+    }
 
-	strcat(ssexchange, serial);
-	strcat(ssexchange, " ");
-	strcat(ssexchange, precedent);
-	strcat(ssexchange, " ");
-	strcat(ssexchange, check);
-	strcat(ssexchange, " ");
-	strcat(ssexchange, section);
+    // ----------------------serial+section--------------------------
+    if (serial_section_mult) {
 
-	return;		// end arrlss
+	checkexchange_serial_section(comment, interactive);
+	return;
     }
 
-    // ----------------------serial+section--------------------------
     if ((serial_section_mult == 1) || (sectn_mult == 1)
```

Comment:
'serial_section_mult' got handled above (line 793). Should be dropped here.

Reply:
It's still work in progress, will drop all unneeded code once ready. It's just kept for reference. Next commits will cover sectn_mult + sectn_mult_once.

Reply:
Was just a reminder from my side...

Reply:
I guess I need to set up for SS and play a bit to see the changes.